test/run_test.py

```diff
@@ -424,21 +430,57 @@ def calculate_shards(num_shards: int, tests: List[str], job_times: Dict[str, Tup
     return sharded_jobs
 
 
-def get_shard(which_shard: int, num_shards: int, tests: List[str]) -> List[str]:
+def pull_job_times_from_S3() -> Dict[str, Tuple[float, int]]:
     if HAVE_BOTO3:
-        s3_reports = get_test_time_reports_from_S3()
+        try:
+            s3_reports = get_test_time_reports_from_S3()
+        except (RuntimeError, RuntimeWarning):
+            print('Ran into issues downloading reports from S3.')
+            s3_reports = []
     else:
-        print('Please install boto3 to enable automatic test sharding.')
+        print('Please install boto3 to enable using S3 test times for automatic sharding and test categorization.')
         s3_reports = []
+
     if len(s3_reports) == 0:
-        print('Gathered no reports from S3. Proceeding with default sharding plan.')
+        print('Gathered no reports from S3. Please proceed without them.')
+        return dict()
+
+    return calculate_job_times(s3_reports)
+
+
+def get_shard(which_shard: int, num_shards: int, tests: List[str]) -> List[str]:
+    jobs_to_times = pull_job_times_from_S3()
+
+    # Got no stats from S3, returning early to save runtime
+    if len(jobs_to_times) == 0:
+        print('Gathered no stats from S3. Proceeding with default sharding plan.')
         return tests[which_shard - 1 :: num_shards]
-    jobs_to_times = calculate_job_times(s3_reports)
+
     shards = calculate_shards(num_shards, tests, jobs_to_times)
     _, tests_from_shard = shards[which_shard - 1]
     return tests_from_shard
 
 
+def get_slow_tests_based_on_S3() -> List[str]:
+    jobs_to_times = pull_job_times_from_S3()
+
+    # Got no stats from S3, returning early to save runtime
+    if len(jobs_to_times) == 0:
+        print('Gathered no stats from S3. Proceeding with default SLOW_TESTS.')
+        return SLOW_TESTS
+
+    new_slow_tests: List[str] = []
+    for test in TESTS:
+        if test in jobs_to_times:
+            test_time, _ = jobs_to_times[test]
+            if test_time > 300:
```

Comment:
300 is just here cuz 5 minutes seemed like a decent threshold. Do we have an actual number though?

Reply:
I believe @walterddr mentioned a threshold of 1 minute (rather than 5)?

Reply:
That's for singular tests, I think, not the entire test file, but it may be the same.

Reply:
Ah you're right, that makes sense

Reply:
We can do that but I do believe a lot of tests in SLOW_TESTS are actually not that slow, like test_determination is 50 seconds (unless that is slow?)

Reply:
in that case we should remove test_determination from the list though.